### ISSUE TITLE
Add space before footnote in stringify function

### ISSUE THREAD

**Issue by lewer** [sergiocorreia/panflute#225]
Hello,

currently, if I have a markdown file like:

```
hello this[^1] is my footnote

[^1]: just a test
```

and if I represent it as a panflute Doc and call stringify, it will return:

`hello thisjust a test is my footnote`

I think that stringify should add a space before the Note element, and instead return:

`hello this just a test is my footnote`

If you agree with this change I can propose a PR.

**Comment by sergiocorreia** [comment]
Sorry for only reading this now. Completely agree with the change, feel free to submit a PR. Otherwise, I'll schedule this for the next major update (time permitting)

**Comment by lewer** [comment]
Actually pandoc's `stringify` function for Lua skips footnotes (see https://github.com/jgm/pandoc/blob/ad3d7c19144b2bac7084499b8fd3a62aa0972a6e/pandoc-lua-engine/src/Text/Pandoc/Lua/Module/Utils.hs#L182)

Maybe panflute's `stringify` should do the same for coherence?

**Comment by sergiocorreia** [comment]
I agree that coherence is valuable but OTOH dropping text might be problematic. Maybe we can just add it back in the same format as [inline footnotes](https://fletcher.github.io/MultiMarkdown-5/footnotes.html) (see last part of this link). EG:

```markdown
hello this[^1] is my footnote

[^1]: just a test
```

Becomse:

```
hello this[^is my footnote]
```

In that way footnotes can be clearly spotted from the main text.

**Comment by lewer** [comment]
But if the footnote contains multiple paragraphs?

I don't think that `stringify`'s goal is to “pretty-print” the element. Do you have any filters in mind for which we would like stringify to return the footnote?

I have in mind filters for which we don't want that. For instance a spellchecking filter: it checks if the element `e` is a Paragraph, and if so, it computes `stringify(e)` and spellchecks. If `stringify` skips the footnote it's ok: as the footnote itself contains a Paragraph, it will be spellchecked separately from `e`. But if `stringify(e)` returns both the paragraph text and the footnote, the footnote will be spellchecked twice, and moreover we'll have to deal with these `[^...]` characters.

what do you think?
